File: rs_test/src/document.rs
#[cfg(test)]
mod document_test {
    use chrono::Utc;
    use draviavemal_openxml_office::document_2007::{Word, WordPropertiesModel};
    use std::fs::{create_dir, exists};

    fn get_save_file(dynamic_path: Option<&str>) -> String {
        let result_path = "test_results";
        if !exists(result_path).expect("Dir Check Failed") {
            create_dir(result_path).expect("Failed to Create")
        }
        format!(
            "{}/test-{}{}.docx",
            result_path,
            dynamic_path.unwrap_or(""),
            Utc::now().format("%Y-%m-%d-%H-%M-%S").to_string()
        )
    }

    #[test]
    fn blank_document() {
        let file = Word::new(None, WordPropertiesModel::default()).expect("Blank File Open Failed");
        file.save_as(&get_save_file(None))
            .expect("Save Result Failed");
        assert_eq!(true, true);
    }

    #[test]
    fn edit_document() {
        let file = Word::new(
            Some("src/tests/TestFiles/basic_test.docx".to_string()),
            WordPropertiesModel::default(),
        )
        .expect("Edit existing file failed");
        file.save_as(&get_save_file(None))
            .expect("Save Result Failed");
        assert_eq!(true, true);
    }
}

File: cs/test/document.cs
// Copyright (c) DraviaVemal. Licensed under the MIT License. See License in the project root.

using draviavemal.openxml_office.global_2007;
using draviavemal.openxml_office.document_2007;

namespace openxmloffice.tests
{
    /// <summary>
    /// Word Test
    /// </summary>
    [TestClass]
    public class Document
    {
        private static readonly string resultPath = "../../test_results";
        /// <summary>
        /// Initialize word Test
        /// </summary>
        /// <param name="context">
        /// </param>
        [ClassInitialize]
        public static void ClassInitialize(TestContext context)
        {
            if (!Directory.Exists(resultPath))
            {
                Directory.CreateDirectory(resultPath);
            }
            PrivacyProperties.ShareComponentRelatedDetails = false;
            PrivacyProperties.ShareIpGeoLocation = false;
            PrivacyProperties.ShareOsDetails = false;
            PrivacyProperties.SharePackageRelatedDetails = false;
            PrivacyProperties.ShareUsageCounterDetails = false;
        }
        /// <summary>
        /// Save the Test File After execution
        /// </summary>
        [ClassCleanup]
        public static void ClassCleanup()
        {
            // word.SaveAs(string.Format("{1}/test-{0}.xlsx", DateTime.Now.ToString("yyyy-MM-dd-HH-mm-ss"), resultPath));
        }

        /// <summary>
        /// 
        /// </summary>
        // [TestMethod]
        public void BlankFile()
        {
            Word word2 = new(new()
            {
                IsInMemory = false
            });
            word2.SaveAs(string.Format("{1}/Blank-{0}.docx", DateTime.Now.ToString("yyyy-MM-dd-HH-mm-ss"), resultPath));
        }

        /// <summary>
        /// Test existing file
        /// </summary>
        [TestMethod]
        public void OpenExistingWord()
        {
            Word word1 = new("test_files/basic_test.docx", new()
            {
                IsInMemory = false
            });
            word1.SaveAs(string.Format("{1}/EditStyle-{0}.docx", DateTime.Now.ToString("yyyy-MM-dd-HH-mm-ss"), resultPath));
            Assert.IsTrue(true);
        }
    }
}
